packages/mobile/src/fiatExchanges/saga.ts

```diff
@@ -121,52 +117,59 @@ function* bidaliPaymentRequest({
   }
 }
 
-export function* searchNewItemsForProviderTxs({ transactions }: NewTransactionsInFeedAction) {
+export function* fetchTxHashesToProviderMapping() {
+  const account = yield call(getAccount)
+  const channel = yield call(providerTxHashesChannel, account)
+
+  try {
+    if (!channel) {
+      throw new Error('Unable to read from providerTxHashesChannel')
+    }
+
+    const txHashesToProvider: TxHashToProvider = yield take(channel)
+    return txHashesToProvider
+  } catch (error) {
+    Logger.error(TAG + 'fetchTxHashesToProviderMapping', error)
+  } finally {
+    if (yield cancelled()) {
+      channel.close()
+    }
+  }
+}
+
+export function* tagTxsWithProviderInfo({ transactions }: NewTransactionsInFeedAction) {
   try {
     if (!transactions || !transactions.length) {
       return
     }
-    Logger.debug(TAG + 'searchNewItemsForProviderTxs', `Checking ${transactions.length} txs`)
 
-    const providerAddresses = yield select(providerAddressesSelector)
-    const lastUsedProvider = yield select(lastUsedProviderSelector)
+    Logger.debug(TAG + 'tagTxsWithProviderInfo', `Checking ${transactions.length} txs`)
+
+    const providerLogos: ProviderLogos = yield select(providerLogosSelector)
+    const txHashesToProvider: TxHashToProvider | undefined = yield call(
+      fetchTxHashesToProviderMapping
+    )
 
     for (const tx of transactions) {
       if (tx.__typename !== 'TokenTransfer' || tx.type !== TokenTransactionType.Received) {
         continue
       }
 
-      if (providerAddresses.includes(tx.address)) {
-        ValoraAnalytics.track(FiatExchangeEvents.cash_in_success, {
-          provider: lastUsedProvider?.name ?? 'unknown',
-        })
-        yield put(assignProviderToTxHash(tx.hash, tx.amount.currencyCode))
+      const provider = txHashesToProvider ? txHashesToProvider[tx.hash] : undefined
```

Comment:
I think this can be simplified to `txHashesToProvider?.[tx.hash]` (likewise for the line below)

Reply:
Oh nice I didn't know you could add the optional chaining when dynamically referencing properties like that